modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/util/BeanUtil.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.util;
+
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Service;
+
+/**
+ * <p>
+ * We cannot inject any Spring-managed bean in any of the provider classes because these are instantiated by JPA before Spring injects anything into it.
+ * This is a utility class to fetch any bean according to our requirements.
+ * </p>
```

Comment:
I'm not sure I'm following. The SPI is a metric source SPI and the current provider is a Graphite provider. I would not expect JPA to load anything related to metrics sources at all.